File: Back-End/Models/GoodTransferDetailModel.cs
using APIBackEnd.Data;
using APIBackend.DataModel;
using System.ComponentModel.DataAnnotations.Schema;
using APIBackEnd.Models;

namespace APIBackend.Models
{
    public class GoodTransferDetailModel
    {
        public int Id { get; set; }
        public int GoodTransferId { get; set; }
        public GoodsTransfer GoodsTransfer { get; set; }
        public int ProductId { get; set; }
        public ProductModel Product { get; set; }
        public int Quantity { get; set; }
    }
}

File: Front-End/angular/src/app/model/goodTransferDetail.model.ts
import { GoodTransfer } from "./goodTransfer.model";
import { Product } from "./product.model";

export class GoodTransferDetails{
    id?: number;
    productId: number;
    quantity: number;
    goodTransferId?: number;
    product?: Product;
    goodTransfer?: GoodTransfer;
}